src/config/config.go

```diff
@@ -42,3 +44,11 @@ func Parse(config []byte) (cfg Config, err error) {
 	}
 	return
 }
+
+func (cfg Config) WriteFile(path string, mode os.FileMode) (err error) {
```

Comment:
This doesn't feel quite right. I'm not sure that WriteFile should exist in this package. Can this be moved into the engine?